crates/flux-middle/src/rty/mod.rs

```diff
@@ -67,6 +67,60 @@ use crate::{
 #[derive(Debug, Clone, Eq, PartialEq, Hash, TyEncodable, TyDecodable)]
 pub struct AdtSortDef(Interned<AdtSortDefData>);
 
+#[derive(Debug, PartialEq, Eq, Hash, TyEncodable, TyDecodable)]
+pub struct AdtSortRefined {
```

Comment:
Maybe we can call this a `SortVariant` and make `AdtSortDef` just a list of these.